samples/Mvc.RazorPagePartial/Pages/Index.cshtml.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+namespace Mvc.RazorPagePartial.Pages
+{
+    public class IndexModel : PageModel
+    {
+        public string Message => DateTime.Now.ToString();
+
+        public IActionResult OnGet() => Page();
+
+        public IActionResult OnGetPartial()
+        {
+            return new PartialViewResult
+            {
+                ViewName = "_IndexPartial",
+                ViewData = new ViewDataDictionary<IndexModel>(this.ViewData, this),
```

Comment:
So this is the magic incantation. Would be nice to have a built-in helper for this of course as this isn't particularly discoverable (passing in two `this` members, plus the current type as the generic arg, eeep).

Reply:
Yup. Basically need https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs#L158-L209 on PageModel